### ISSUE TITLE
[BUG] logrotate config waagent-extn.logrotate conflict with azure-lad

### ISSUE THREAD

**Issue by taoyama** [Azure/WALinuxAgent#1804]
`/etc/logrotate.d/waagent-extn.logrotate` which is installed by WALinuxAgent 2.2.46 conflict with `/etc/logrotate.d/azure-lad` which is installed by lad-mdsd-1.5.109-LADmaster.101.x86_64.

**Distro and WALinuxAgent details (please complete the following information):**
 - Any Distro with 2.2.46

**Additional context**
The rotate file was merged by #1740 

**Log file attached**

```
# logrotate /etc/logrotate.conf
error: waagent-extn.logrotate:1 duplicate log entry for /var/log/azure/Microsoft.Azure.Diagnostics.LinuxDiagnostic/extension.log
```

**Comment by taoyama** [comment]
https://github.com/Azure/azure-linux-extensions/issues/1045

**Comment by narrieta** [comment]
@pgombar  Agent update apparently also overwrites log rotate settings for waagent.log

**Comment by sbonds** [comment]
The specific issue we see is when updating the RPMs, the file `/etc/logrotate.d/waagent.logrotate` was replaced with a default config.

In my opinion, the issue I see isn't a WALinuxAgent github issue, it's a RedHat RPM packaging issue and the problematic SPEC doesn't seem to have a public version control system for filing issue

**Comment by sbonds** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1838254 filed for the above

**Comment by pgombar** [comment]
@sbonds I believe @taoyama's issue is different.

@sbonds, do you also see this error?
```
# logrotate /etc/logrotate.conf
error: waagent-extn.logrotate:1 duplicate log entry for /var/log/azure/Microsoft.Azure.Diagnostics.LinuxDiagnostic/extension.log
```

@taoyama could you confirm if the [reported bug ](https://bugzilla.redhat.com/show_bug.cgi?id=1838254) is different from the issue you're seeing?

**Comment by taoyama** [comment]
@pgombar the reported bug is about rpm packaging issue. That is different from the issue I am seeing.

**Comment by Poil** [comment]
Hi,

It looks like I have the same bug on Ubuntu 20.04
```
Nov 04 00:00:01 dpl-uat-vms-100900000C logrotate[41321]: error: waagent-extn.logrotate:1 duplicate log entry for /var/log/azure/Microsoft.Azure.Diagnostics.LinuxDiagnostic/extension.log
Nov 04 00:00:01 dpl-uat-vms-100900000C logrotate[41321]: error: found error in file waagent-extn.logrotate, skipping
Nov 04 00:00:01 dpl-uat-vms-100900000C systemd[1]: logrotate.service: Main process exited, code=exited, status=1/FAILURE
```

**Comment by johan-chassaing** [comment]
Hi there!

I think that I found the same issue:

```
# /usr/sbin/logrotate -s /var/lib/logrotate/logrotate.status /etc/logrotate.conf
error: waagent-extn.logrotate:1 duplicate log entry for /var/log/azure/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux/extension.log
error: found error in file waagent-extn.logrotate, skipping
```

It's looks like, the waagent try to handle the log rotation for omsagent logs but omsagent has already a logrorate configuration:

```
omsagent-02463909-facf-4e68-85af-10fa3f9a193c:/var/log/azure/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux/watcher.log {
omsagent-02463909-facf-4e68-85af-10fa3f9a193c:/var/log/azure/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux/extension.log {
waagent-extn.logrotate:/var/log/azure/*/*.log
```

Is waagent is legitimate to rotate the logs of other plugin/extension which are not populated by waagent ?

**Comment by narrieta** [comment]
Fixed by #2143
